src/providers/DevicesProvider/AudioInputProvider.tsx

```diff
@@ -67,16 +79,18 @@ const AudioInputProvider: React.FC = ({ children }) => {
           console.log(
             'Previously selected audio input lost. Selecting a default device.'
           );
-          meetingManager.selectAudioInputDevice(newAudioInputs[0].deviceId);
+          currentDevice = await replaceDevice(newAudioInputs[0].deviceId);
+          await meetingManager.selectAudioInputDevice(currentDevice);
```

Comment:
So we're basically saying - pass in your own implementation for selecting a default device? Do we have demo changes to showcase the correct way to do this?

Reply:
Or can we provide like a default implementation that they can override?

Reply:
Here is the demo change: https://github.com/aws-samples/amazon-chime-sdk/pull/70/files

Builders can implement their own `onDeviceReplacement` method to create a transform device when audio devices change or just ignore this, then we will pass a normal device instead. 

Just realized that, not only when default device changes, we need to call this `onDeviceReplacement` function, also when the selected device is disconnected.